mender/templates/tests/test-mender.yaml

```
@@ -0,0 +1,46 @@
+{{- if .Values.tests.enabled }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ include "mender.fullname" . }}-test-mender
+  labels:
+    {{- include "mender.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['/bin/sh', '-c']
+      args:
+        {{- if and (.Values.global.enterprise) (.Values.tenantadm.enabled) (.Values.auditlogs.enabled) }}
+        - echo "Starting smoketests";
+          echo "Testing deployment service";
+          wget -qO- '{{ include "mender.fullname" . }}-deployments:{{ .Values.deployments.service.port }}/api/internal/v1/deployments/health';
+          echo "Testing tenantadm service";
+          wget -qO- '{{ include "mender.fullname" . }}-tenantadm:{{ .Values.tenantadm.service.port }}/api/internal/v1/tenantadm/tenants';
+          echo "Testing auditlogs service";
+          wget -qO- '{{ include "mender.fullname" . }}-auditlogs:{{ .Values.deployments.service.port }}/api/internal/v1/auditlogs/health';
+          echo "Testing deviceauth service";
+          wget -qO- '{{ include "mender.fullname" . }}-device-auth:{{ .Values.device_auth.service.port }}/api/internal/v1/devauth/health';
+          echo "Testing deviceconfig service";
+          wget -qO- '{{ include "mender.fullname" . }}-deviceconfig:{{ .Values.deviceconfig.service.port }}/api/internal/v1/deviceconfig/health';
+          echo "Testing deviceconnect service";
+          wget -qO- '{{ include "mender.fullname" . }}-deviceconnect:{{ .Values.deviceconnect.service.port }}/api/internal/v1/deviceconnect/health';
+          echo "Testing inventory service";
+          wget -qO- '{{ include "mender.fullname" . }}-inventory:{{ .Values.inventory.service.port }}/api/internal/v1/inventory/health';
+          echo "Testing iot_manager service";
+          wget -qO- '{{ include "mender.fullname" . }}-iot-manager:{{ .Values.iot_manager.service.port }}/api/internal/v1/iot-manager/health';
+          echo "Testing useradm service";
+          wget -qO- '{{ include "mender.fullname" . }}-useradm:{{ .Values.useradm.service.port }}/api/internal/v1/useradm/users';
```

Why not use the `/health` endpoint?
```suggestion
          wget -qO- '{{ include "mender.fullname" . }}-useradm:{{ .Values.useradm.service.port }}/api/internal/v1/health';
```

---

Because the health endpoint is actually already checked by the health probe. The `/useradm/users` is an "application check", so it's even better than the health, in the smoke test scope

---

The same applies to tests for the other services. OTOH, there is some logic behind the `/health` endpoints (it pings the databases and other depending services).